starknet_devnet/starknet_wrapper.py

```diff
@@ -165,16 +143,15 @@ async def call_or_invoke(self, choice: Choice, specifications: InvokeFunction):
 
         return { "result": adapted_result }, execution_info
 
-    def __is_transaction_hash_legal(self, transaction_hash_int: int) -> bool:
-        return 0 <= transaction_hash_int < len(self.__transaction_wrappers)
-
     def get_transaction_status(self, transaction_hash: str):
         """Returns the status of the transaction identified by `transaction_hash`."""
 
-        transaction_hash_int = int(transaction_hash, 16)
+        if transaction_hash in self.__transaction_wrappers:
```

Comment:
Here transaction_hash is a string I assume, but __transaction_wrappers is Dict[int, TransactionWrapper]. The initial implementation had ints as keys because I wanted to avoid issues with leading zeros (e.g. "0x01" and "0x1" not being the same hash), so that's why I did all the conversions to int.